### PR TITLE
Fix #3548 - allow hyphens as well in panel id

### LINKED ISSUE #3548: Allow for hyphens in "Panel ID"
When creating a new gene panel from the user interface, the internal field "Panel ID" allows for underscores but not for hyphens ("bindestreck", see attached image). For reasons beyond my understanding (perhaps @mhkc can fill in), underscores don't work in our pipeline in Lund, so our bioinformaticians have changed to hyphens in the backend. With increasing use of gene panels, however, we would appreciate if an end user could assign Panel ID with hyphens from start and reduce unnecessary work load on our bioinformaticians. In summary, if there are no technical counter arguments, please allow for hyphens in the field "Panel ID"!

Panel ID

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Test in particular that there are no downstream ill effects of hyphens like dots - compare #2400.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN

### CODE REVIEW

**Comment by dnil** [comment]
Ok, looks good so far:
Screenshot 2022-08-25 at 13 06 27

and no obvious issues like with the dot names:
Screenshot 2022-08-25 at 13 07 41

File: scout/server/blueprints/panels/templates/panels/panels.html [hunk 1]
```diff
@@ -40,7 +47,7 @@
         <div class="row">
           <div class="col-5">
             <input type="text" name="search_for" class="form-control typeahead_gene mb-1" data-provide="typeahead" autocomplete="off" placeholder="Search Gene in Panels" required
-                   pattern="^[0-9]+\s*\|\s*.*" title="Only alphanumeric characters (A-Z+a-z+0-9) and underscores allowed.">
+                   pattern="^[0-9]+\s*\|\s*.*" title="Search for gene symbol. Type name or number and the field autocompletes a full description. Allowed patterns have a numerical hgnc id and a pipe character.">
```
Comment (northwestwitch) [inline scout/server/blueprints/panels/templates/panels/panels.html:50]: 👍🏻

File: scout/server/blueprints/panels/templates/panels/panels.html [hunk 2]
```diff
@@ -109,7 +116,7 @@
           </div>
           <div class="col-sm-3 text-center">
               <input type="text" name="new_panel_name" class="form-control" placeholder="Panel ID" required
-                pattern="^[a-zA-Z0-9_]*$" title="Only alphanumeric characters (A-Z+a-z+0-9) and underscores allowed.">
+                pattern="^[a-zA-Z0-9_-]*$" title="Only alphanumeric characters (A-Z+a-z+0-9), hyphens, and underscores allowed.">
```
Comment (northwestwitch) [inline scout/server/blueprints/panels/templates/panels/panels.html:119]: 👍🏻

**Review by northwestwitch — APPROVED** [review]
Looks good!

**Pull request merged**